packages/protocol/contracts/governance/Election.sol

```diff
@@ -845,4 +891,52 @@ contract Election is
     }
     return res;
   }
+
+  struct RevokeVotesInfo {
+    address[] groups;
+    uint256 remainingValue;
+  }
+
+  /**
+   * @notice Reduces the total amount of `account`'s voting gold by `value` by
+   *         iterating over all groups voted for by account.
+   * @param account Address to revoke votes from.
+   * @param value Amount of votes to revoke.
+   * @param lessers The groups receiving fewer votes than the i'th `group`, or 0 if
+   *                the i'th `group` has the fewest votes of any validator group.
+   * @param greaters The groups receivier more votes than the i'th `group`, or 0 if
+   *                the i'th `group` has the most votes of any validator group.
+   * @param indices The indices of the i'th group in the account's voting list.
+   */
+  function forceRevokeVotes(
+    address account,
+    uint256 value,
+    address[] calldata lessers,
+    address[] calldata greaters,
+    uint256[] calldata indices
+  ) external nonReentrant onlyRegisteredContract(LOCKED_GOLD_REGISTRY_ID) returns (uint256) {
+    require(value > 0 && value <= this.getTotalVotesByAccount(account));
+    RevokeVotesInfo memory info = RevokeVotesInfo(votes.groupsVotedFor[account], value);
+    require(
+      info.groups.length == lessers.length &&
+        lessers.length == greaters.length &&
+        greaters.length == indices.length
+    );
+    // Iterate in reverse order to hopefully optimize removing pending votes before active votes
+    for (uint256 i = info.groups.length; i > 0; i = i.sub(1)) {
+      info.remainingValue = info.remainingValue.sub(
+        _revokeVotes(
+          account,
+          info.groups[i.sub(1)],
+          info.remainingValue,
+          lessers[i.sub(1)],
+          greaters[i.sub(1)],
+          indices[i.sub(1)]
+        )
+      );
+      if (info.remainingValue == 0) {
+        return value;
```

Comment:
Should we break here instead and return `value.sub(info.remainingValue)`? Right now nothing is returned outside of the loop